api/src/modules/indicator-records/indicator-records.service.ts

```diff
@@ -156,69 +154,70 @@ export class IndicatorRecordsService extends AppBaseService<
    */
   // TODO still not adapted to modular indicators, because the performance gets hit drastically on the source import. Pending to be worked on
   async createIndicatorRecordsForAllSourcingRecords(): Promise<void> {
+    const indicatorMap: Record<string, any> = await this.getIndicatorMap();
+    // Using a hardcoded list of indicator types because this function is still based on
+    // a calculation query with hardcoded indicators,
+    const indicators: INDICATOR_TYPES[] = [
+      INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
+      INDICATOR_TYPES.CARBON_EMISSIONS,
+      INDICATOR_TYPES.DEFORESTATION,
+      INDICATOR_TYPES.BIODIVERSITY_LOSS,
+    ];
+
     const rawData: SourcingRecordsWithIndicatorRawDataDto[] =
       await this.indicatorRecordRepository.getIndicatorRawDataForAllSourcingRecords();
-    const calculatedData: IndicatorRecordCalculatedValuesDto[] = rawData.map(
-      (sourcingRecordData: SourcingRecordsWithIndicatorRawDataDto) =>
-        this.calculateIndicatorValues(sourcingRecordData),
+
+    const calculatedData2: IndicatorRecordCalculatedValuesDto[] = rawData.map(
+      (sourcingRecordData: SourcingRecordsWithIndicatorRawDataDto) => {
+        // Small DTO transformations to adapt the result of the stored procedures query
+        const indicatorValues: Map<INDICATOR_TYPES, number> = new Map();
+        indicatorValues.set(
+          INDICATOR_TYPES.DEFORESTATION,
+          sourcingRecordData.rawDeforestation,
+        );
+        indicatorValues.set(
+          INDICATOR_TYPES.CARBON_EMISSIONS,
+          sourcingRecordData.rawCarbon,
+        );
+        indicatorValues.set(
+          INDICATOR_TYPES.BIODIVERSITY_LOSS,
+          sourcingRecordData.rawBiodiversity,
+        );
+        indicatorValues.set(
+          INDICATOR_TYPES.UNSUSTAINABLE_WATER_USE,
+          sourcingRecordData.rawWater,
+        );
+
+        const indicatorComputedRawDataDto: IndicatorComputedRawDataDto = {
+          harvestedArea: sourcingRecordData.harvestedArea,
+          production: sourcingRecordData.production,
+          indicatorValues,
+        };
+
+        return this.calculateIndicatorValuesV2(
```

Comment:
Can't we effectively refactor this to 
```suggestion
        return this.calculateIndicatorValues(
```
And remove the 'old' approach?